mlfinlab/data_structures/run_data_structures.py

```diff
@@ -90,45 +93,55 @@ def _extract_bars(self, data):
 
             if imbalance > 0:
                 self.imbalance_array['buy'].append(imbalance)
-                # Set zero to keep buy and sell arrays synced
-                self.imbalance_array['sell'].append(0)
                 cum_theta_buy += imbalance
+                buy_ticks += 1
             elif imbalance < 0:
                 self.imbalance_array['sell'].append(abs(imbalance))
-                # Set zero to keep buy and sell arrays synced
-                self.imbalance_array['buy'].append(0)
                 cum_theta_sell += abs(imbalance)
 
-            if not list_bars and np.isnan(self.exp_buy_proportion):
-                self.exp_buy_proportion, self.exp_sell_proportion = self._get_expected_imbalance(
-                    self.exp_num_ticks, self.imbalance_array)
+            if not list_bars and np.isnan([self.exp_imbalance['buy'], self.exp_imbalance['sell']]).any():
```

Comment:
Can we add a comment to explain what is happening here? 

* Can we assign this bool to a named variable? np.isnan([self.exp_imbalance['buy'], self.exp_imbalance['sell']]).any()

Reply:
Not a big deal. You decide.